sync-online/index.js

```diff
@@ -0,0 +1,68 @@
+// Our textarea
+var textarea = document.querySelector('textarea');
+
+// Register the ServiceWorker
+if (navigator.serviceWorker) {
+  navigator.serviceWorker.register('service-worker.js', {
+    scope: '.'
+  })
+  .then(function(registration) {
+    console.log('The service worker has been registered ', registration, registration.active);
+
+    if (navigator.serviceWorker.controller && navigator.serviceWorker.controller.state === 'activated') {
+      init();
+    } else {
+      // Listen for claiming of our ServiceWorker
+      navigator.serviceWorker.addEventListener('controllerchange', function(event) {
+        console.log('[controllerchange] A "controllerchange" event has happened within navigator.serviceWorker: ', event);
+
+        // Listen for changes in the state of our ServiceWorker
+        navigator.serviceWorker.controller.addEventListener('statechange', function() {
+          console.log('[controllerchange][statechange] A "statechange" has occured: ', this.state);
+
+          // If the ServiceWorker becomes "activated", let the user know they can go offline!
+          if (this.state === 'activated') {
+            init();
+          }
+        });
+      });
+    }
+  })
+  .catch(function() {
+    console.error('The service worker could not be registered');
+    init();
+  });
+} else {
+  init();
+}
+
+// Function to get content for page
+function init() {
+  return fetch('?get')
+    .then(function(response) {
+      return response.json();
```

Comment:
Seeing this error:

```
// Error:
// Uncaught (in promise) SyntaxError: Unexpected token o
// Because the fetch is seeing "[object Object]""
```

But I'm sending back:

```
return new Response({
            ok: true,
            status: '200',
            statusText: 'OK',
            headers: { 'content-type': 'application/json' },
            body: JSON.stringify(content)
          });
```
... so I'm confused.